nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -688,8 +725,15 @@ contract StakingEscrow is Issuer {
             policyManager.register(_staker, getCurrentPeriod() - 1);
         }
         token.safeTransferFrom(_payer, address(this), _value);
-        info.value += _value;
+        uint256 newValue = info.value += _value;
         lock(_staker, _value, _periods);
+
+        if(!info.flags.bitSet(SNAPSHOTS_DISABLED_INDEX)){
+            info.history.addSnapshot(newValue);
+            uint256 lastGlobalBalance = uint256(balanceHistory.lastValue());
+            balanceHistory.addSnapshot(lastGlobalBalance + newValue);
+        }
```

Comment:
I guess we can extract this block to internal function, something like `addSnapshot(StakerInfo storage _info, int256 _addition)`, and this will decrease size of contract
Also question about `newValue`, why not just `info.value`?

Reply:
✅